processes/code-styles/swift/swift-style-guide-en.md

```diff
@@ -1,59 +1,269 @@
-  * [Struct Initializers](#struct-initializers)
-  * [CGRect Functions](#cgrect-functions)
-  * [Lazy Initialization](#lazy-initialization)
-  * [Type Inference](#type-inference)
-  * [Syntactic Sugar](#syntactic-sugar)
-* [Functions vs Methods](#functions-vs-methods)
-* [Memory Management](#memory-management)
-  * [Extending Lifetime](#extending-lifetime)
-* [Access Control](#access-control)
-* [Control Flow](#control-flow)
-* [Golden Path](#golden-path)
-  * [Failing Guards](#failing-guards)
-* [Semicolons](#semicolons)
-* [Parenthesis](#parenthesis)
-* [Credits](#credits)
-
-
-## Correctness
-
-Consider warnings to be errors. This rule informs many stylistic decisions such as not to use the `++` or `--` operators, C-style for loops, or strings as selectors.
-
-## Naming
-
-Descriptive names with camel case for classes, methods, variables, etc **MUST** be used. Type names (classes, structures, enumerations, protocols, typealiases) **MUST** be capitalized, while everything else **MUST** start with a lower case letter.
+  * [Table of Contents](#table-of-contents)
+  * [1. File structure](#1-file-structure)
+  * [2. Code Formatting](#2-code-formatting)
+  * [3. Naming](#3-naming)
+  * [4. Code Organization](#4-code-organization)
+  * [5. Coding Style](#5-coding-style)
+    + [5.1 General](#51-general)
+    + [5.2 Access Modifiers](#52-access-modifiers)
+    + [5.3 Custom Operators](#53-custom-operators)
+    + [5.4 Switch Statements and `enum`s](#54-switch-statements-and-enums)
+    + [5.5 Optionals](#55-optionals)
+    + [5.6 Properties](#56-properties)
+    + [5.7 Closures](#57-closures)
+    + [5.8 Arrays](#58-arrays)
+    + [5.9 Error Handling **TODO**](#59-error-handling-todo)
+    + [5.10 Using `guard` Statements](#510-using-guard-statements)
+    + [5.11 Classes and Structures. Which one to use?](#511-classes-and-structures-which-one-to-use)
+    + [5.12 Types](#512-types)
+    + [5.13 Struct Initializers](#513-struct-initializers)
+  * [5.14 Functions vs Methods](#514-functions-vs-methods)
+  * [5.15 Selectors](#515-selectors)
+  * [5.16 Correctness](#516-correctness)
+  * [6. Documentation and Comments](#6-documentation-and-comments)
+    + [6.1 Documentation](#61-documentation)
+    + [6.2 Comments](#62-comments)
+  * [7. Credits](#7-credits)
+
+## 1. File structure
+
+* **2.11** Filenames should be named after type they contain. Don't use abbreviations.
+
+**ConnectionTableViewCell.swift**
+
+```swift
+class ConnectionTableViewCell: UITableViewCell {
+}
+```
+
+**MyOwnProtocol.swift**
+
+```swift
+protocol MyOwnProtocol {
+}
+```
+
+* **2.12** File folder structure should reflect XCode project folder structure. XCode 9 will do it for you by default when adding new files to a project.
+
+
+
+## 2. Code Formatting
+
+* **2.11** Use 4 spaces for tabs.
+* **2.12** Avoid uncomfortably long lines with a hard maximum of 160 characters per line (Xcode->Preferences->Text Editing->Page guide at column: 160 is helpful for this)
```

Comment:
Should we add screenshot here?

Reply:
Don't think so. I think we can even remove the path to this setting at all. It's not hard to find it but direct path or screenshot may lead to the wrong place if Xcode settings UI will change.